app/src/main/java/io/zenandroid/onlinego/utils/Globals.kt

```diff
@@ -44,8 +41,8 @@ fun egfToRank(rating: Double?) =
 fun formatRank(rank: Double?) =
         when(rank) {
             null -> "?"
-            in 0 until 30 -> if (!SettingsRepository.hideRanks) "${ceil(30 - rank).toInt()}k" else ""
-            in 30 .. 100 -> if (!SettingsRepository.hideRanks) "${ceil(rank - 29).toInt()}d" else ""
+            in 0 until 30 -> if (SettingsRepository.showRanks) "${ceil(30 - rank).toInt()}k" else ""
```

Comment:
I'm afraid this in not the proper place to implement this. While this way we have the minimum number of changes to implement the feature, it does increase the responsibility of this function. Furthermore it does not allow you to display the ranks in some places (maybe I want to see my own ratings in the stats, but not the opponent's rating in the game).

Instead, I suggest we check this at the View level: something like `rankTextView.showIf(SettingsRepository.showRanks)`.

Reply:
I thought about this - the problem is that it implies plenty of views (game, cards, player search, opponent picker...) and is easier to forget when adding new views. Also some of the formatRank uses/rank displays don't have their own views either (e.g. opponent search/picker)

OGS hides it everywhere, but does have a specific "show your rating" button only for the stats screen - how about doing the same in the app?

Reply:
True, it implies editing a fair few views, but such is the life of a developper. Otherwise it will be quite possible to forget you have that check there and later wonder why is that method not working as you would expect it too.

To be honest I think we can start by hiding it on the main screen and on the game screen and then call it a day.